my-dsl.md

```diff
@@ -2,37 +2,78 @@
 _What is the name of the language? Link the name to its webpage 
 (if appropriate)._
 
+I looked at the language [Cg][1].
+
+[1]: https://developer.nvidia.com/cg-toolkit
+
 
 # Domain
 _Describe the language's domain in five words._
 
+vertex/pixel shading for GPUs
+
 
 # Computational model
 _What is the underlying computational model of this language? To answer this 
 question, provide a high-level description (no more than 100 words) of the 
 computation that occurs when someone executes a program in this language._
 
+According to the [Cg tutorial][2], section 1.4.2, Cg is executed to interact
+with the OpenGL or Direct3D API for your GPU. The way this works is for a Cg
+program to be translated into OpenGL or Direct3D, then for the translated
+program to be sent to the GPU using Open GL or Direct3D, and finally for OpenGL
+or Direct3D to execute it on the GPU.
+
+[2]: http://http.developer.nvidia.com/CgTutorial/cg_tutorial_chapter01.html
+
 
 # DSL-ness
 _Fowler writes about a spectrum of languages, from general-purpose languages to 
 "purely" domain-specific. Where does the DSL you chose fall on this spectrum, 
-and why?_ 
+and why?_
+
+Cg is somewhere in the middle. It is intended to be used for GPU operations,
+but it includes so much of C that it could be used more generally. As Fowler
+says on page 30, "A particular usage of a language can put it on either side of
+the DSL line." I would lean towards Cg being more domain-specific, simply
+because there is little reason to use it instead of C unless you are doing GPU
+operations. See ["Cg in Two Pages"][3] for a summary of differences with C.
+
+[3]: http://arxiv.org/pdf/cs/0302013v1.pdf
 
 
 # Internal or external?
 _Is the language implemented as an internal or external DSL? 
 Justify your answer._
 
+Cg is an example of an internal DSL. Its syntax and data types are based
+on C, with some important differences highlighted on NVIDIA's
+[language specification][4].
+
+[4]: http://http.developer.nvidia.com/Cg/Cg_language.html
+
 
 # Host language
 _What language(s) was (were) used to implement the DSL?_
 
+Cg is hosted in C.
+
 
 # Benefits
 _Identify one potential benefit of the DSL: how is a programmer's life or a 
 company's bottom line made easier by the existence of this language?_
 
+Cg makes it much easier and cleaner to write code for vertex and fragment
+shaders. The other main option was assembly, which is not so nice! Writing
+Cg code (which is essentially C code) would be significantly more productive.
+
```

Comment:
I know nothing about writing code for vertex or fragment shaders but anything is better than assembly it would seem. Do you know why they stopped supporting Cg?